test/functional/test_framework/ellsq.py

```diff
@@ -94,4 +94,69 @@ def r(x,y,i):
     if y.is_odd() == u.is_odd():
         return u
     else:
-        return -u
\ No newline at end of file
+        return -u
+
+def SECP256K1_FE_CONST(d7, d6, d5, d4, d3, d2, d1, d0):
+    n = []
+    n.append(d0 | (((d1) & 0xFFFFF) << 32))
+    n.append((d1 >> 20) | (d2 << 12) | ((d3 & 0xFF) << 44))
+    n.append((d3 >> 8) | ((d4 & 0xFFFFFFF) << 24))
+    n.append((d4 >> 28) | (d5 << 4) | ((d6 & 0xFFFF) << 36))
+    n.append((d6 >> 16) | (d7 << 16))
+    sum = 0
+    for i in range(5):
+        sum += n[i]*(2**(i*52))
+    return fe(sum)
+
+def SECP256K1_GE_CONST(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p):
+    fe1 = SECP256K1_FE_CONST(a, b, c, d, e, f, g, h)
+    fe2 = SECP256K1_FE_CONST(i, j, k, l, m, n, o, p)
+    return (fe1, fe2)
+
+ellsq_tests = [
+    [SECP256K1_GE_CONST(0xc27fb7a3, 0x283a7d3e, 0xc9f96421, 0x545ef6f5, 0x8ace7b71, 0x06c8a1b9, 0x07c0ae8a, 0x7598159c, 0xe05a060e, 0x839ef79f, 0xc0c1267c, 0xa17880c9, 0x584cdd34, 0xc05f9695, 0x55482207, 0xe6851f2a), [SECP256K1_FE_CONST(0xc0ad127a, 0xa36824d6, 0x5b1f5be7, 0x4de1aa25, 0xbc4d5cbe, 0xcee15462, 0x0a12682a, 0xfc87df98), SECP256K1_FE_CONST(0xd40fd5bc, 0x51992484, 0x8f13273b, 0x1d857cba, 0x42d45e78, 0x9eaa4e47, 0xf458b83a, 0xbd5f8d1c), SECP256K1_FE_CONST(0xde636141, 0x7deb440b, 0x3a305924, 0x43635cf9, 0xcf42f9b5, 0xf5b891c1, 0x1e119f09, 0x71b570ac), SECP256K1_FE_CONST(0xd55135ce, 0x41bb4d05, 0x5b3757f4, 0xaf1d6537, 0x137376d7, 0x5270caae, 0xda68382d, 0x25d00708)]],
```

Comment:
I would avoid the round-tripping through the awkward secp256k1 8x32 notation for field elements.

and then for the first line (just concatenating the hex digits, dropping the `, 0x` between them).

```python
    [(fe(0xc27fb7a3283a7d3ec9f96421545ef6f58ace7b7106c8a1b907c0ae8a7598159c), fe(0xe05a060e839ef79fc0c1267ca17880c9584cdd34c05f969555482207e6851f2a)), [fe(0xc0ad127aa36824d65b1f5be74de1aa25bc4d5cbecee154620a12682afc87df98), fe(0xd40fd5bc519924848f13273b1d857cba42d45e789eaa4e47f458b83abd5f8d1c), fe(0xde6361417deb440b3a30592443635cf9cf42f9b5f5b891c11e119f0971b570ac), fe(0xd55135ce41bb4d055b3757f4af1d6537137376d75270caaeda68382d25d00708)]],
```

etc.

Reply:
Thanks! This looks really nice.